RingPublishingTracking/src/main/java/com/ringpublishing/tracking/internal/decorator/WindowSizeDecorator.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ *  Created by Grzegorz Małopolski on 10/6/21, 11:28 AM
+ * Copyright © 2021 Ringier Axel Springer Tech. All rights reserved.
+ *
+ */
+
+package com.ringpublishing.tracking.internal.decorator
+
+import android.content.Context
+import android.os.Build
+import android.view.WindowInsets
+import android.view.WindowManager
+import com.ringpublishing.tracking.data.Event
+
+internal class WindowSizeDecorator(context: Context) : DeviceScreenDecorator(context)
+{
+
+	private val windowManager = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
+
+	override fun decorate(event: Event)
+	{
+		event.add(EventParam.WINDOW_SIZE, getWindowSizeString())
+	}
+
+	private fun getWindowSizeString(): String = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R)
```

Comment:
czym jest R i jak to sie ma do min targetu etc?

Reply:
R to jest android 11.
Od 11 mamy nowy system mierzenia wysokości, bo też jest wprowadzone multi window i okno może zmieniać rozmiar.

Reply:
a wcześniej nie może? ja mam na tablecie chyba nizsza wersje systemu

Reply:
Sprawdziłem dokładniej i multiwindow jest od android api 30 ale za to od android 7 a w sumie android 9 jest możliwość zrobienia czegoś podobnego czyli split screen. Dodałem obsługę tego, choć nakombinowałem się pół dnia jak to zrobić dobrze. Wymaga to zmiany w inicjalizacji modułu przekazujemy Application zamiast context.